tilequeue/queue/writer.py

```diff
@@ -0,0 +1,47 @@
+# coordinates all the pieces required to enqueue coordinates
+
+
+class QueueWriter(object):
+
+    def __init__(self, tile_queue, coord_grouper, msg_marshaller,
+                 inflight_manager, enqueue_batch_size):
+        self.tile_queue = tile_queue
+        self.coord_grouper = coord_grouper
+        self.msg_marshaller = msg_marshaller
+        self.inflight_manager = inflight_manager
+        self.enqueue_batch_size = enqueue_batch_size
+
+    def _enqueue_batch(self, queue_send_data_chunk):
+        payloads = []
+        all_coords = []
+        # used for metadata
+        zooms = []
+        for payload, coords, metadata in queue_send_data_chunk:
+            payloads.append(payload)
+            all_coords.extend(coords)
+            zoom = metadata.get('zoom')
+            assert zoom is not None
+            zooms.append(zoom)
+
+        metadata = dict(zooms=zooms)
```

Comment:
I'm confused about what's going on here. The coords have been grouped, each one getting a single `zoom` in the metadata, then we batch the groups, adding the list of zooms as metadata? It looks like they just get `izip`ped back together in the SQS queue writer.

Reply:
Yea, it's a bit confusing/convoluted at the moment. The initial grouping is by zoom level to serialize the payload, and then the multisqs is grouping again based on the queue that it will send to, which is also based on zoom. I think this also points to removing the multisqs queue that I was hinting at above, in favor of having all the grouping happen in one place. I think I'll just move in that direction and I'm hoping that the code will become simpler here.